cloud/amazon/ec2_customer_gateway.py

```diff
@@ -208,19 +217,22 @@ def main():
     name = module.params.get('name')
 
     existing = gw_mgr.describe_gateways(module.params['ip_address'])
+    # describe_gateways returns a key of CustomerGateways where as create_gateway returns a
+    # key of CustomerGateway. For consistency, change it here
+    existing['CustomerGateway'] = existing['CustomerGateways']
```

Comment:
The way it has been coded further on always points to the first element of the list.  This is OK because boto3s response syntax is a list of dicts (and there will always be only 1 element in the list).

If I made your suggested change then I'd have to remove all the `[0]` references further down so i think it's best to leave as is.